### ISSUE TITLE
Nginx won't display alerta 

### ISSUE THREAD

**Issue by jgoetz** [alerta/alerta-webui#262]
I am using http://docs.alerta.io/en/latest/gettingstarted/tutorial-1-deploy-alerta.html#tutorial-1 to install Alerta on a standalone CentOS7 under VMWare. I have DNS set up to point alerta.dri.edu to this server. 

At the end of Step 2,  when I visit http://alerta.dri.edu Nginx displays the default "Welcome to CentOS" page. 

I'm sure it's a configuration problem, but I can't find where I went wrong. 

Here's my configuration files, etc. 

```
[root@alerta nginx]# ls -la /var/www/html
total 396
drwxr-xr-x 8 nginx nginx   4096 Dec 17 13:07 .
drwxr-xr-x 4 nginx nginx     67 Dec 17 13:55 ..
-rw-r--r-- 1 root  root     197 Dec 17 07:08 404.html
-rw-r--r-- 1 root  root    7612 Dec 17 07:08 apple-touch-icon-114x114.png
-rw-r--r-- 1 root  root    5136 Dec 17 07:08 apple-touch-icon-120x120.png
-rw-r--r-- 1 root  root    4768 Dec 17 07:08 apple-touch-icon-144x144.png
-rw-r--r-- 1 root  root    8361 Dec 17 07:08 apple-touch-icon-152x152.png
-rw-r--r-- 1 root  root    1796 Dec 17 07:08 apple-touch-icon-57x57.png
-rw-r--r-- 1 root  root    1800 Dec 17 07:08 apple-touch-icon-60x60.png
-rw-r--r-- 1 root  root    1795 Dec 17 07:08 apple-touch-icon-72x72.png
-rw-r--r-- 1 root  root    2992 Dec 17 07:08 apple-touch-icon-76x76.png
drwxr-xr-x 2 root  root      38 Dec 17 07:08 audio
-rw-r--r-- 1 root  root      14 Dec 17 07:08 CNAME
-rw-r--r-- 1 root  root      21 Dec 17 11:49 config.json
-rw-r--r-- 1 root  root      38 Dec 17 07:08 config.json.example
drwxr-xr-x 2 root  root     115 Dec 17 07:08 css
drwxr-xr-x 6 nginx nginx   4096 Dec 13 14:09 dist1
-rw-r--r-- 1 root  root    1186 Dec 17 07:08 favicon-128.png
-rw-r--r-- 1 root  root     241 Dec 17 07:08 favicon-16x16.png
-rw-r--r-- 1 root  root   14155 Dec 17 07:08 favicon-196x196.png
-rw-r--r-- 1 root  root     319 Dec 17 07:08 favicon-32x32.png
-rw-r--r-- 1 root  root    1597 Dec 17 07:08 favicon-96x96.png
-rw-r--r-- 1 root  root   34494 Dec 17 07:08 favicon.ico
drwxr-xr-x 2 root  root     191 Dec 17 07:08 fonts
-rw-r--r-- 1 root  root    3034 Dec 17 07:08 index.html
drwxr-xr-x 2 root  root     284 Dec 17 07:08 js
-rw-r--r-- 1 root  root    4768 Dec 17 07:08 mstile-144x144.png
-rw-r--r-- 1 root  root   35835 Dec 17 07:08 mstile-150x150.png
-rw-r--r-- 1 root  root   57463 Dec 17 07:08 mstile-310x150.png
-rw-r--r-- 1 root  root  131257 Dec 17 07:08 mstile-310x310.png
-rw-r--r-- 1 root  root    1186 Dec 17 07:08 mstile-70x70.png
drwxr-xr-x 7 root  root    4096 Dec 13 14:09 old_dist
-rw-r--r-- 1 root  root    3034 Dec 17 11:49 temp.html
```

My wsgi.py 
```
[root@alerta nginx]# cat /var/www/wsgi.py
from alerta import app
```

My Nginx configuration. I put the code from the installation instructions in a file named default and placed it in conf.d instead of sites-enabled, because on CentOS the installer creates the conf.d folder instead of the sites-enabled and sites-available folders.
```
[root@alerta nginx]# cat /etc/nginx/nginx.conf
# For more information on configuration, see:
#   * Official English Documentation: http://nginx.org/en/docs/
#   * Official Russian Documentation: http://nginx.org/ru/docs/

user nginx;
worker_processes auto;
error_log /var/log/nginx/error.log;
pid /run/nginx.pid;

# Load dynamic modules. See /usr/share/doc/nginx/README.dynamic.
include /usr/share/nginx/modules/*.conf;

events {
    worker_connections 1024;
}

http {
    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                      '$status $body_bytes_sent "$http_referer" '
                      '"$http_user_agent" "$http_x_forwarded_for"';

    access_log  /var/log/nginx/access.log  main;

    sendfile            on;
    tcp_nopush          on;
    tcp_nodelay         on;
    keepalive_timeout   65;
    types_hash_max_size 2048;

    include             /etc/nginx/mime.types;
    default_type        application/octet-stream;

    # Load modular configuration files from the /etc/nginx/conf.d directory.
    # See http://nginx.org/en/docs/ngx_core_module.html#include
    # for more information.
    include /etc/nginx/conf.d/*.conf;

#    server {
#        listen       80 default_server;
#        listen       [::]:80 default_server;
#        server_name  _;
#        root         /usr/share/nginx/html;
#
#        # Load configuration files for the default server block.
#        include /etc/nginx/default.d/*.conf;
#
#        location / {
#        }
#
#        error_page 404 /404.html;
#            location = /40x.html {
#        }
#
#        error_page 500 502 503 504 /50x.html;
#            location = /50x.html {
#        }
#    }
#

# Settings for a TLS enabled server.
#
#    server {
#        listen       443 ssl http2 default_server;
#        listen       [::]:443 ssl http2 default_server;
#        server_name  _;
#        root         /usr/share/nginx/html;
#
#        ssl_certificate "/etc/pki/nginx/server.crt";
#        ssl_certificate_key "/etc/pki/nginx/private/server.key";
#        ssl_session_cache shared:SSL:1m;
#        ssl_session_timeout  10m;
#        ssl_ciphers HIGH:!aNULL:!MD5;
#        ssl_prefer_server_ciphers on;
#
#        # Load configuration files for the default server block.
#        include /etc/nginx/default.d/*.conf;
#
#        location / {
#        }
#
#        error_page 404 /404.html;
#            location = /40x.html {
#        }
#
#        error_page 500 502 503 504 /50x.html;
#            location = /50x.html {
#        }
#    }

}
```

Here's the conf.d folder contents showing the default file listed, and then the contents of that file. 
```
[root@alerta nginx]# ls -la /etc/nginx/conf.d/
total 8
drwxr-xr-x. 2 root root   21 Dec 17 14:40 .
drwxr-xr-x. 4 root root 4096 Dec 17 14:40 ..
-rw-r--r--  1 root root  494 Dec 17 14:35 default

[root@alerta nginx]# cat /etc/nginx/conf.d/default
server {
        listen 80 default_server;
        listen [::]:80 default_server;

        location /api { try_files $uri @api; }
        location @api {
            include uwsgi_params;
            uwsgi_pass unix:/tmp/uwsgi.sock;
            proxy_set_header Host $host:$server_port;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }

        location / {
                root /var/www/html;
        }
}
```
Here is my config.json file and location:
```
[root@alerta nginx]# cat /var/www/html/dist/config.json
{"endpoint":"/api"}
```
Here's my uwsgi service declaration. I verified wsgi is working by serving a test page from the command line.
```
[root@alerta nginx]# cat /etc/systemd/system/uwsgi.service
[Unit]
Description=uWSGI service

[Service]
ExecStart=/opt/alerta/bin/uwsgi --ini /etc/uwsgi.ini

[Install]
WantedBy=multi-user.target
```

My wsgi ini file.
```
[root@alerta nginx]# cat /etc/uwsgi.ini
[uwsgi]
chdir = /var/www
mount = /api=wsgi.py
callable = app
manage-script-name = true
env = BASE_URL=/api

master = true
processes = 5
logger = syslog:alertad

socket = /tmp/uwsgi.sock
chmod-socket = 664
uid = nginx
gid = nginx
vacuum = true

die-on-term = true
```
And finally my alertad.conf file and location:
```
[root@alerta nginx]# cat /etc/alertad.conf
SECRET_KEY='ZvBkn$E7S$6n5CZ#k4z^Yp4fC@D7mGabX!yj!@rE'
BASE_URL='/api'
AUTH_REQUIRED=True
ADMIN_USERS=['jgoetz@dri.edu']
DEBUG=True
LOGGER_NAME='alerta'
LOG_FILE='/var/log/alerta.log'

SEVERITY_MAP = {
    'fatal': 0,
    'critical': 1,
    'warning': 4,
    'indeterminate': 5,
    'ok': 5,
    'unknown': 9
}
DEFAULT_NORMAL_SEVERITY = 'ok'
COLOR_MAP = {
    'severity': {
        'fatal': 'blue',
        'critical': 'red',
        'warning': '#1E90FF',
        'indeterminate': 'lightblue',
        'ok': '#00CC00',
        'unknown': 'silver'
    }
}
PLUGINS=['reject']
ALLOWED_ENVIRONMENTS=['Production', 'Development', 'Code']

CORS_ORIGINS = [
    'http://localhost',
    'http://localhost:8000',
    r'https?://\w*\.?local\.alerta\.io:?\d*/?.*'  # => http(s)://*.local.alerta.io:<port>
]
```

Can you please take a look and tell me what I've done wrong? Thanks!!

**Comment by satterly** [comment]
Rename `etc/nginx/conf.d/default` to `etc/nginx/conf.d/default.conf` and restart nginx.

**Comment by jgoetz** [comment]
Thanks, Nick. I appreciate your time on such a trivial question. 
Unfortunately, I now get the "ERROR: Failed to connect to Alerta API due to missing or invalid config.json file." error when I try http://alerta.dri.edu.  

I added the dri line to the  /etc/alertad.conf:
CORS_ORIGINS = [
    'http://localhost',
    'http://alerta.dri.edu',
    'http://alerta.dri.edu:8080',
    'https?://\w*\.?local\.alerta\.io:?\d*/?.*' 
]

I have rebooted my server to make sure I didn't have any remnant processes working on old configurations. I'm still getting the same error. 

The developers tools don't show any errors when retrieving pages (status 200) except for GET localhost:8080 config xhr request doesn't get a response.  

When I try alerta.dri.edu/api I get a "Problem loading page" error, and developers tools shows the response as "blocked" under timings. 

I can test uwsgi with a dummy page and it works. What else could I be doing wrong?
Thanks again for your time.
